### ISSUE TITLE
Cannot share projects on Chromebooks

### ISSUE THREAD

**Issue by chrisgarrity** [LLK/scratchjr#449]
### Expected Behavior

When you try to share a project by clicking the 'share by email' button you should see a share sheet to select the application you would like to use to share the project. The share sheet should include email, files and drive if available.

### Actual Behavior

A message appears at the top of the screen that "No apps can perform this action"

### Device and Operating System

HP x2 Chromebook

**Comment by yueyuzhao** [comment]
Hi @chrisgarrity , I guess the intent type which is wrongly set that causes the issue, could you please give it a shot that change the file `JavaScriptDirectInterface.java` on line 656 from `it.setType(mimetype);` to `it.setType("message/rfc822");`?

But you said this does not appear to be an issue using the old version, so it is maybe also not relevant. If this doesn't work, I will try to get a chromebook or setup a test environment first. 
Thanks.

**Comment by chrisgarrity** [comment]
@yueyuzhao that's definitely in the right vicinity. Instead of "No apps..." it opens Chrome with a Gmail compose message. Unfortunately the connection between the Android app running on Chrome and Gmail in the browser loses the attached file (this has always been true). So instead of explicitly opening an email mimetype we need one that will pop open the normal Android share sheet and include options like gmail, drive, files etc.

**Comment by chrisgarrity** [comment]
@yueyuzhao I started to try to figure out where the change was introduced. However I ended up with inconsistent results when installing the app on a couple of Chromebooks I have at home. On one of the Chromebooks I installed the new internal test version of ScratchJr (1.2.12), and I got the problem when trying to export. Then I noticed it needed to restart to update Chrome. After the restart, exporting in the internal test version works fine. The other Chromebook I have is running in developer mode, and it still has the problem with no apps being found. 

In the end I think we should wait to see if this appears on other normal Chromebooks - I have some internal testing with other Scratch Team members scheduled. If they don't see the problem I'll close this issue.

**Comment by yueyuzhao** [comment]
Hi @chrisgarrity , thank you for trying out and sorry for this not working attempt. FYI, I got this "No apps ...." message once because I didn't set up the email account in the email app, I can do it normally after the set up. 

Also I can still share the project as attachment after change the intent type to `"message/rfc822"` on my Android tablets. 

One question is on iOS if the user want share the project to computer or to files, he/she needs to click the `Airdrop` button, and when clicking the email button, the user can share the project only via Email. Why don't we implement the same strategy on Android if we need the user to save the project to drive or files?

**Comment by chrisgarrity** [comment]
@yueyuzhao sorry for the false alarm! I had powerwashed the chromebook recently and none of the Android versions of the apps were installed. After I went to the play store and installed the Android versions of Gmail and Drive they showed up as options for export. We actually need to do a full refactor (design and implementation) of the export/import UI. So much has changed since it was implemented in 2015.
